packages/@aws-cdk/aws-rds/lib/proxy.ts

```diff
@@ -0,0 +1,429 @@
+
+  /**
+   * The secret that the proxy uses to authenticate to the RDS DB instance or Aurora DB cluster.
+   * These secrets are stored within Amazon Secrets Manager.
+   *
+   * @default - no secret
+   */
+  readonly secret: secretsmanager.ISecret;
+
+  /**
+   * One or more VPC security groups to associate with the new proxy.
+   *
+   * @default - No security groups
+   */
+  readonly securityGroups?: ec2.ISecurityGroup[];
+
+  /**
+   * The subnets used by the proxy.
+   *
+   * @default - the VPC default strategy if not specified.
+   */
+  readonly vpcSubnets?: ec2.SubnetSelection;
+
+  /**
+   * The VPC to associate with the new proxy.
+   */
+  readonly vpc: ec2.IVpc;
+}
+
+/**
+ * Construction properties for a DatabaseProxy
+ */
+export interface DatabaseProxyProps extends DatabaseProxyOptions {
+  /**
+   * DB proxy target: Instance or Cluster
+   */
+  readonly proxyTarget: ProxyTarget
+}
+
+/**
+ * Properties that describe an existing DB Proxy
+ */
+export interface DatabaseProxyAttributes {
+  /**
+   * DB Proxy Name
+   */
+  readonly dbProxyName: string;
+
+  /**
+   * DB Proxy ARN
+   */
+  readonly dbProxyArn: string;
+
+  /**
+   * Endpoint
+   */
+  readonly endpoint: string;
+
+  /**
+   * The security groups of the instance.
+   */
+  readonly securityGroups: ec2.ISecurityGroup[];
+}
+
+/**
+ * DB Proxy
+ */
+export interface IDatabaseProxy extends cdk.IResource {
+  /**
+   * DB Proxy Name
+   *
+   * @attribute
+   */
+  readonly dbProxyName: string;
+
+  /**
+   * DB Proxy ARN
+   *
+   * @attribute
+   */
+  readonly dbProxyArn: string;
+
+  /**
+   * Endpoint
+   *
+   * @attribute
+   */
+  readonly endpoint: string;
+}
+
+/**
+ * RDS Database Proxy
+ *
+ * @resource AWS::RDS::DBProxy
+ */
+export class DatabaseProxy extends cdk.Resource
+  implements IDatabaseProxy, ec2.IConnectable, secretsmanager.ISecretAttachmentTarget {
+  /**
+   * Import an existing database proxy.
+   */
+  public static fromDatabaseProxyAttributes(
+    scope: cdk.Construct,
+    id: string,
+    attrs: DatabaseProxyAttributes,
+  ): IDatabaseProxy {
+    class Import extends cdk.Resource implements IDatabaseProxy {
+      public readonly dbProxyName = attrs.dbProxyName;
+      public readonly dbProxyArn = attrs.dbProxyArn;
+      public readonly endpoint = attrs.endpoint;
+    }
+    return new Import(scope, id);
+  }
+
+  /**
+   * DB Proxy Name
+   *
+   * @attribute
+   */
+  public readonly dbProxyName: string;
+
+  /**
+   * DB Proxy ARN
+   *
+   * @attribute
+   */
+  public readonly dbProxyArn: string;
+
+  /**
+   * Endpoint
+   *
+   * @attribute
+   */
+  public readonly endpoint: string;
+
+  /**
+   * Access to network connections.
+   */
+  public readonly connections: ec2.Connections;
+
+  protected readonly resource: CfnDBProxy;
+
+  constructor(scope: cdk.Construct, id: string, props: DatabaseProxyProps) {
+    super(scope, id, { physicalName: props.dbProxyName || id });
+
+    const role = new iam.Role(this, 'IAMRole', {
```

Comment:
> Enabling TLS by default is a common security best practice. Any particular reason it should be different here?

Rather than forcing it, it is better to let users choose when they really need it.

In my case, I won't use TLS on db proxy because I have no profit for it now.

```
new rds.DatabaseProxy(stack, 'proxy', { ... });
```

looks better than

```
new rds.DatabaseProxy(stack, 'proxy', { ..., requireTLS: false });
```

Reply:
We want customers to get the best security posture out of the box where possible. Let's default to enabling TLS here.